File: server/Models/Vault.cs
namespace server.Models
{
    public class Vault
    {

        public int id { get; set; }
        public string creatorId { get; set; }
        public string name { get; set; }
        public string description { get; set; }
        public bool isPrivate { get; set; }
        public Profile creator { get; set; }
    }

    public class VaultViewModel : Vault
    {
        public int VaultKeepId { get; set; }
    }
}

File: client/src/models/Vault.js
export class Vault {
  constructor(data = {}) {
    this.id = data.id || ''
    this.name = data.name || ''
    this.isPrivate = data.isPrivate
    this.description = data.description || ''
    this.img = data.img || '//placehold.it/400x400'
    this.creator = data.creator || { name: 'unknown', picture: '//placehold.it/20x20' }
  }
}
